### PR TITLE
Add script to automate production deployment

### PR DESCRIPTION
**Pull request by bglopez**
This is in reference to issue #248 -- it foregoes the env_prod.list file since docker-compose seems to have issues with parsing in the environment variables and instead uses a BASH script that takes input to replace the variables in a templated docker-prod.yml file. This is more of a proof-of-concept than something I expect to be production ready.

### CODE REVIEW

File: docker-prod-template.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  api:
+    image: openeats/api
+    command: ./base/gunicorn_start.sh
+    restart: on-failure
+    volumes:
+      - static-files:/code/static-files
+      - site-media:/code/site-media
+    depends_on:
+      - db
+    environment:
```
Comment (RyanNoelk) [inline docker-prod-template.yml:12]: The api needs to be aware of the following env vars as well:
- MYSQL_DATABASE
- MYSQL_ROOT_PASSWORD

The api needs them in order to login to the database and perform I/O

**Review by RyanNoelk — COMMENTED** [review]
I'll take a closer look at this in the next few days. But the env file needs to be aware of more values than what you have listed in the docker-compose file. IMO, it is much easier to manage a single env file for all the contains than multiple ones for each different container.

**Comment by RyanNoelk** [comment]
Taking a better look at this I really like the idea. My only problem with is the lack of use of the env file. IMO I think we need to find a way to use the file as it's much easier to maintain and would make deploying simpler.

**Comment by bglopez** [comment]
Hi Ryan,

Sorry for the delayed response. I just ran a test to grab the environment variable from within the container using the env file and it appears to be set correctly there:

```
# python
Python 2.7.13 (default, Jul 24 2017, 20:13:18)
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> print(os.environ.get('ALLOWED_HOST'))
'food.neverlocate.me'
>>>
```

but the settings.py file still does not contain the host in the allowed hosts list. I can modify the script to work with an ENV file since I agree that it would be the best way to go about this.

**Comment by bglopez** [comment]
Using a quick test Python script I do see that the quotes may be an issue, but it still doesn't explain why settings.py isn't getting the correct variable:

```
 python test.py
ALLOWED_HOSTS before append: ['localhost', '127.0.0.1', 'node']
env_allowed_host: 'food.neverlocate.me'
ALLOWED_HOSTS after append: ['localhost', '127.0.0.1', 'node', "'food.neverlocate.me'"]
env_allowed_host: 'food.neverlocate.me'
```

test.py
```
import os

ALLOWED_HOSTS = ['localhost', '127.0.0.1', 'node']

env_allowed_host = os.environ.get('ALLOWED_HOST', None)
if env_allowed_host is not None:
    print 'ALLOWED_HOSTS before append: %s' % (ALLOWED_HOSTS)
    print 'env_allowed_host: %s' % (env_allowed_host)
    ALLOWED_HOSTS.append(env_allowed_host)
    print 'ALLOWED_HOSTS after append: %s'  % (ALLOWED_HOSTS)
    print 'env_allowed_host: %s' % (env_allowed_host)
```

Is this something that may require settings.py to be modified before docker-compose builds and not use the pre-built image?

**Comment by RyanNoelk** [comment]
The settings file shouldn't have anything to do with the image being built vs using a pre-built image. When the container first starts up and tries to either run gunicorn or the django dev server, it will load the settings files and read the environment vars.

If you try and change the environment vars after the container has started then the settings files (and the django app) won't have those changes. You need to restart the containers if you do make a change to the environment file.

You can add something like this to the settings file to see what it's getting when it's booting the app.

``` python 
ALLOWED_HOSTS = ['localhost', '127.0.0.1', 'node']

print("allowed Host list:")
print(ALLOWED_HOSTS)

print("allowed Host ENV:")
print(os.environ.get('ALLOWED_HOST', None))

env_allowed_host = os.environ.get('ALLOWED_HOST', None)
if env_allowed_host is not None:
    ALLOWED_HOSTS.append(env_allowed_host)

print("allowed Host list after append:")
print(ALLOWED_HOSTS)
```

Then restart you docker-compose with:
Note: we are printing the all logging to stdout here.
```
docker-compose stop
docker-compose up
```
You should see something like this:
```
api_1   | allowed Host list:
api_1   | ['localhost', '127.0.0.1', 'node']
api_1   | allowed Host ENV:
api_1   | ryannoelk.ocm
api_1   | allowed Host list after append:
api_1   | ['localhost', '127.0.0.1', 'node', 'ryannoelk.ocm']
```

**Comment by RyanNoelk** [comment]
Closing this PR for now. I have added a script to start the containers and the containers will now automatically create static content and apply DB migrations.

**Pull request closed without merging**